src/collidable.ts

```diff
@@ -0,0 +1,41 @@
+import { PhysObject } from './physObject';
+import { CanvasValues } from './constants';
+
+export function willCollide(
+  object: PhysObject,
+  objects: Array<PhysObject>
+): Array<PhysObject> {
+  const results: Array<PhysObject> = [];
+  for (const collidable of objects) {
+    if (
+      collidable.x + collidable.width / 2 > object.x - object.width / 2 &&
+      collidable.x - collidable.width / 2 < object.x + object.width / 2 &&
+      collidable.y + collidable.height / 2 > object.y - object.height / 2 &&
+      collidable.y - collidable.height / 2 < object.y + object.height / 2
+    ) {
+      results.push(collidable);
+    }
+  }
+  return results;
+}
+
+export function borderCollision(object: PhysObject) {
```

Comment:
I'd recommend naming anything that returns a specific thing as "getThatThing" for clarity. Also, consider adding a return type.

Reply:
Answered in other conversation. Went with getCanvasEdgeCollision.